checks.d/zk.py

```diff
@@ -68,22 +128,32 @@ def check(self, instance):
             else:
                 status = AgentCheck.WARNING
             message = u'Response from the server: %s' % ruok
-        self.service_check('zookeeper.ruok', status, message=message,
-                           tags=sc_tags)
+        finally:
+            self.service_check('zookeeper.ruok', status, message=message,
+                    tags=sc_tags)
 
         # Read metrics from the `stat` output.
         try:
             stat_out = self._send_command('stat', *cx_args)
         except ZKConnectionFailure:
             self.increment('zookeeper.timeouts')
+            self.report_instance_mode(hostname, 'down', tags)
+            raise
+        except:
+            e = sys.exc_info()[1]
+            print >> sys.stderr, "Error: %s" % e
```

Comment:
You should perhaps catch the base Exception with `except Exception as e` and then log it via `self.warning`. That way it shows up with the `dd-agent info` command.